### PR TITLE
Adds Anti-Gravity Grenades to Black Market Uplink

### PR DESCRIPTION
**Pull request by desertmote**
## About The Pull Request

Adds anti-gravity grenades to the BMU.

## Why It's Good For The Game

A well-suited item for combat with security officers, as the anti grav gives immunity to snares.

## Changelog
:cl:
add: Anti-Gravity Grenades to Black Market Uplink
/:cl:

### CODE REVIEW

**Comment by Admiral-Gigaton** [comment]
The ablative baseball bat, that's the one that deflects lasers, right? Like the IRS pirates, their kneecapper has a chance to deflect them, yeah?

**Comment by honkpocket** [comment]
> The ablative baseball bat, that's the one that deflects lasers, right? Like the IRS pirates, their kneecapper has a chance to deflect them, yeah?

This bat has a 100% chance to block laser projectiles.

**Comment by EuclidianGeometer** [comment]
saying these are additions from novastation requested to be moved upstream isn't actually explaining why it's good for the game.

**Comment by necromanceranne** [comment]
I've spent the last few years removing total energy weapon immunity from the game, and yet here we are. Not just trying to add another one to the game, but to let literally anyone with 2000 credits just straight up buy it? Not even an antagonist weapon or nothing?

Whoever told you to PR this upstream was clearly pulling a prank on you.

(Also, adding admin-spawn rarities as obtainable in-game has usually been frowned upon in the past anyways)

**Comment by Rex9001** [comment]
Perhaps a version of the bat with less or no laser block would work better? On TG not even the dsword or energy shield have full laser block

**Comment by honkpocket** [comment]
> Perhaps a version of the bat with less or no laser block would work better? On TG not even the dsword or energy shield have full laser block

Of course, but the energy-swords hit a lot harder than a bat.

**Comment by Rex9001** [comment]
> > On TG not even the dsword or energy shield have full laser block
> 
> 
> 
> Of course, but the energy-swords hit a lot harder than a bat.

Yeah but an item which doesnt hit at all whos only function is to block, blocks worse than this baseball bat

**Comment by Admiral-Gigaton** [comment]
near complete and total immunity to lasers is a little busted for a normal dude to get, even an ERT without admin assistance couldnt get you, and a deathsquad has like maybe 12 rounds each of 356, which is a lot of damage, but is hard to hit you with, to kill you.

nerf the block chance a bit (giving people a counter to sec brutality is always cool, but maybe not 100%) and then its pretty sweet

**Comment by honkpocket** [comment]
How about I make it drain a small amount of stamina per reflect.

**Comment by Admiral-Gigaton** [comment]
> How about I make it drain stamina per reflect.

great idea, and it could promote working with people in chemistry more, or using plumbing to do more than mass produce burn and brute meds to bypass surgery, to then make stamina meds. could also promote more use of implants and stun reduction surgeries, so medbay and robotics sees more use.

**Comment by jughu** [comment]
stun reducing surgeries is useless for stamcrit they only reduce paralyze

**Comment by Admiral-Gigaton** [comment]
> stun reducing surgeries is useless for stamcrit they only reduce paralyze

well, implants and chems still get attention, and so will cargo (acquiring cash to purchase, acquiring gear to go with it because clearly if you have it you're using it)

**Comment by jughu** [comment]
theres no implant that increases stam regen only paralyze

Only chems will get attenion but dunno if thats something we want aswell, curreny the state of chem dispensers are very shit they usually are drained 5-10 minutes into roundstart and it forces medical to either get a cell charger and a screwdriver crowbar constantly reconstructing the thing or a inducer, ideally we want less tiders breaking into departments for their powergame.potentional

Would it possible to make the baseball bat way more skill based like auctally clicking the projectiles themselves to reflect them back if its.possible atleast.

**Comment by Admiral-Gigaton** [comment]
> theres no implant that increases stam regen only paralyze

no dropping implant, so if you do get stun, sec cant take your precious super weapon

**Comment by Likteer** [comment]
@necromanceranne  You can reflect thrown items with a baseball bat if you time it right and hit it mid-air. Why not change the metal bat to allow the same for lasers instead of passive protection?

**Comment by jughu** [comment]
> > theres no implant that increases stam regen only paralyze
> 
> no dropping implant, so if you do get stun, sec cant take your precious super weapon

They still can though. it only makes it so you don't drop it on paralyze so you can still abuse blockchance

**Comment by Exester509** [comment]
> @necromanceranne You can reflect thrown items with a baseball bat if you time it right and hit it mid-air. Why not change the metal bat to allow the same for lasers instead of passive protection?

Giving the bat an energy projectile parry would be pretty hilarious honestly

**Comment by necromanceranne** [comment]
> @necromanceranne You can reflect thrown items with a baseball bat if you time it right and hit it mid-air. Why not change the metal bat to allow the same for lasers instead of passive protection?

I'm fairly sure the ablative bat already does that, though you would have to ask fikou. That's much better than straight up passive defensive power.

**Comment by Jacquerel** [comment]
I suspect the difference between "by holding this you can reflect infinite lasers" and "by holding this you can reflect 20 lasers" is not likely to be very significant.

**Comment by honkpocket** [comment]
> I suspect the difference between "by holding this you can reflect infinite lasers" and "by holding this you can reflect 20 lasers" is not likely to be very significant.

I anticipated the DragNET and disabler SMG winning from this bat to be sufficient, to be honest.

**Comment by Admiral-Gigaton** [comment]
> I suspect the difference between "by holding this you can reflect infinite lasers" and "by holding this you can reflect 20 lasers" is not likely to be very significant.

almost every shift that ive seen as sec, the warden or sec buys a or several crates of disabler SMGs, so im pretty sure its gunna be fine. worst case people start spamming melee battles and using batons as they should be using batons, best case its balanced and forces john shitcurity to stop dragnet snaring all the time, and instead try its alternative, teleporting a tider into the middle of the furthest Z level from the station

**Comment by Jacquerel** [comment]
I'm simply not interested in adding a 20 force weapon which also allows you to no-sell an entire disabler clip or more than one laser gun clips, sorry. 

The fact that stamina regenerates rapidly after you stop taking stamina damage essentially nullifies any downside of using this except if people dump an enormous amount of shots into you, it functionally isn't a downside at all.

I think I'd really only be satisfied if this requires some kind of active component, slowed you for a period of time after activating, or had at least like four times the stamina cost that it does now. Or any other actually meaningful downside you could add, but frankly it takes quite a big one to justify 100% projectile reflection.

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 1]
```diff
@@ -55,10 +55,13 @@
...
 				span_userdanger("The [bullet.name] gets reflected by [src]!"),
 			)
-			// Finds and plays the block_sound of item which reflected
 			for(var/obj/item/held_item in held_items)
+				// Plays the block_sound of the item which reflected
 				if(held_item.IsReflect(def_zone))
 					playsound(src, held_item.block_sound, BLOCK_SOUND_VOLUME, TRUE)
+				// Applies a stamina penalty if the reflect item asks for it
+				if(held_item.IsReflectCostly())
+					apply_damage(REFLECT_STAMINA_COST, STAMINA)
```
Comment (ATH1909) [inline code/modules/mob/living/carbon/human/human_defense.dm:64]: hey so there are these cool things called hulks

Comment (honkpocket) [reply code/modules/mob/living/carbon/human/human_defense.dm:64]: Hello! I know hulks exist, my preferred method of take-down is a mutadone syringe supplied by the CMO.
I don't see how this is relevant, sorry. Using security gear to try and take down a hulk is really questionable and a strange stance to use for arguing.

Comment (moocowswag) [reply code/modules/mob/living/carbon/human/human_defense.dm:64]: You cannot expect security to fully rely on another department to take down a hulk, not to mention hulk's power disparity cannot be that high considering its a mutation that only humans can have.

To allow hulks to use something that synergizes with them is to buff hulks and by extension humans.

Comment (honkpocket) [reply code/modules/mob/living/carbon/human/human_defense.dm:64]: Just remove hulk. That's all I have left to add to this.

**Comment by honkpocket** [comment]
Let me know if you want this nerf to stay in the PR, considering this baseball bat already spawns in-round as part of a pirate-ship. Or if you want to keep the anti-gravity grenade. If neither, I'll just go ahead and close this.

**Comment by Jacquerel** [comment]
If you want to PR this as just the anti-gravity grenade and just leave the bat on the pirate ship as it is (that's still pretty strong, but it's at least only for an antagonist), that'd be fine by me.

File: code/modules/mob/living/carbon/human/human_defense.dm (hunk 1 above)
Comment (Iamgoofball) [reply code/modules/mob/living/carbon/human/human_defense.dm:64]: she's actually cooking here removing hulk is good

**Comment by Iamgoofball** [comment]
@Jacquerel lemme hit u with some wisdom: the baseball bat is imho fine, if it's locked to having to two-hand it to do the reflection that'd make it perfectly balanced
bolas let you get in to melee with a baton or something, the Shotgun Don't Miss, thrown weapons exist even if security barely uses 'em, can't reflect pepperspray, can't reflect flashbangs, if it's two handed you can't even chuck 'em back all you can do is try to run

**Comment by Xackii** [comment]
> I'm simply not interested in adding a 20 force weapon which also allows you to no-sell an entire disabler clip or more than one laser gun clips, sorry.
> 
> The fact that stamina regenerates rapidly after you stop taking stamina damage essentially nullifies any downside of using this except if people dump an enormous amount of shots into you, it functionally isn't a downside at all.
> 
> I think I'd really only be satisfied if this requires some kind of active component, slowed you for a period of time after activating, or had at least like four times the stamina cost that it does now. Or any other actually meaningful downside you could add, but frankly it takes quite a big one to justify 100% projectile reflection.



> I'm simply not interested in adding a 20 force weapon which also allows you to no-sell an entire disabler clip or more than one laser gun clips, sorry.
> 
> The fact that stamina regenerates rapidly after you stop taking stamina damage essentially nullifies any downside of using this except if people dump an enormous amount of shots into you, it functionally isn't a downside at all.
> 
> I think I'd really only be satisfied if this requires some kind of active component, slowed you for a period of time after activating, or had at least like four times the stamina cost that it does now. Or any other actually meaningful downside you could add, but frankly it takes quite a big one to justify 100% projectile reflection.

create a black market uplink, roll a baseball bat in the black market uplink, find 1000+ credits to buy it, search for the bat all over the station hoping it hasn't been stolen, and get a reflector armor that occupies the hand slot, doesn't provide blocking chances, and consumes 5 stamina (which never recovers with server ticks), making this item more like a pillow than an abuse of game mechanics. Why bother with all this as a traitor when you can just steal reflective armor or order it from cargo?"

**Comment by san7890** [comment]
i noticed you're trying to rebrand this PR but i think you should clone/copy your branch and just create a fully new PR about adding anti-gravity grenades to the uplink as this would have been staled out if not for shitposters as it's on the complete bottom of the PR tracker.

just copy over the `BMU-adds` branch into a new one and just start from scratch, the problem is any new discourse is already polluted with people complaining about the bat. a new PR will get a fresh set of calm eyes to give reasoned feedback instead of whatever was going on above. thanks.

**Pull request closed without merging**